zone/spells.cpp

```diff
@@ -5520,6 +5520,26 @@ uint32 Client::GetHighestScribedSpellinSpellGroup(uint32 spell_group)
 	return highest_spell_id;
 }
 
+uint32 Client::GetHighestSpellinSpellGroup(uint32 spell_group)
+{
+	//Typical live spells follow 1/5/10 rank value for actual ranks 1/2/3, but this can technically be set as anything.
+
+	int highest_rank = 0; //highest ranked found in spellgroup
+	uint32 highest_spell_id = 0;  //spell_id of the highest ranked spell
+
+	for (int i = 0; i < EQ::spells::SPELL_ID_MAX; i++) {
+		if (IsValidSpell(i)) {
+			if (spells[i].spell_group == spell_group) {
+				if (highest_rank < spells[i].rank) {
+					highest_rank = spells[i].rank;
+					highest_spell_id = i;
+				}
+			}
+		}
+	}
+	return highest_spell_id;
+}
+
```

Comment:
Consolidation similar to above.
```cpp
uint32 Client::GetHighestSpellinSpellGroup(uint32 spell_group)
{
	//Typical live spells follow 1/5/10 rank value for actual ranks 1/2/3, but this can technically be set as anything.

	int highest_rank = 0; //highest ranked found in spellgroup
	uint32 highest_spell_id = 0;  //spell_id of the highest ranked spell

	for (uint16 i = 0; i < SPDAT_RECORDS; i++) {
		if (
			IsValidSpell(i) &&
			spells[i].spell_group == spell_group &&
			highest_rank < spells[i].rank
		) {
			highest_rank = spells[i].rank;
			highest_spell_id = i;
		}
	}

	return highest_spell_id;
}
```

Reply:
consolidated